File: StdCore.Web/StdCore.Web.LoadersScheduler/ClientApp/src/models/task.model.ts
import moment from "moment";
import {ITaskEntity} from "../data/task";
import {IUserEntity} from "../data/user";
import {TaskStatus} from "../data/enum/task-status";
import {TaskType} from "../data/enum/task-type";
import {TaskDirectionType} from "../data/enum/task-direction-type";
import {IUserModel} from "./user.model";

export interface ITaskModel {
    id: number,
    direction: TaskDirectionType,
    loaderId: number,
    planStartUtc: number,
    planEndUtc: number,
    factStartUtc: number,
    factEndUtc: number,
    createdUtc: number,
    status: TaskStatus,
    type: TaskType,
    comment: string,
    createUser: IUserModel
}

export class TaskModel implements Omit<ITaskModel, "planStartUtc" | "planEndUtc" | "factStartUtc" | "factEndUtc"> {
    comment: string;

    factEnd: moment.Moment;
    factStart: moment.Moment;
    planEnd: moment.Moment;
    planStart: moment.Moment;
    createUser: IUserEntity;

    id: number;
    loaderId: number;
    status: TaskStatus;
    type: TaskType;
    direction: TaskDirectionType;

    constructor() {
        this.planStart = moment();
        this.planEnd = moment().add(30, "minutes");
    }

    createdUtc: number;
}

File: StdCore.Web/StdCore.Web.LoadersScheduler/Models/TaskModel.cs
using System;
using System.Text.Json.Serialization;
using System.Threading.Tasks;
using Std.Common.DateTime;
using Std.WebClient.Data.Enum;

namespace Std.WebClient.Models
{
    public class TaskModel
    {
        public int Id { get; set; }
        [JsonIgnore]
        public TaskDirection Direction { get; set; }
        [JsonPropertyName("Direction")]
        public string DirectionName { get; set; }
        public int? LoaderId { get; set; }
        [JsonIgnore]
        public DateTime PlanStart { get; set; }
        [JsonIgnore]
        public DateTime PlanEnd { get; set; }
        [JsonIgnore]
        public DateTime? FactStart { get; set; }
        [JsonIgnore]
        public DateTime? FactEnd { get; set; }
        [JsonIgnore]
        public DateTime Created { get; set; }
        [JsonIgnore]
        public TaskStatus Status { get; set; }
        [JsonPropertyName("Status")]
        public string StatusName { get; set; }
        public TaskType Type { get; set; }
        [JsonPropertyName("Type")]
        public string TypeName { get; set; }
        public string Comment { get; set; }
        public int CreatedUserId { get; set; }
        public UserModel CreateUser { get; set; }
        public long PlanStartUtc
        {
            get => PlanStart.ToUnixTimeStamp();
            set => PlanStart = value.UnixTimeStampToDateTime();
        }
        public long PlanEndUtc
        {
            get => PlanEnd.ToUnixTimeStamp();
            set => PlanEnd = value.UnixTimeStampToDateTime();
        }
        public long? FactStartUtc
        {
            get => FactStart?.ToUnixTimeStamp();
            set => FactStart = value?.UnixTimeStampToDateTime();
        }
        public long? FactEndUtc
        {
            get => FactEnd?.ToUnixTimeStamp();
            set => FactEnd = value?.UnixTimeStampToDateTime();
        }
    }
}


